src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EcDsa.ImportExport.cs

```diff
@@ -0,0 +1,286 @@
+                    if (curveType == ECCurve.ECCurveType.Characteristic2)
+                    {
+                        // Match Windows semantics where a,b,gx,gy,qx,qy have same length
+                        // Treat length of m separately as it is not tied to other fields for Char2 (Char2 not supported by Windows) 
+                        cbFieldLength = GetMax(new[] { a_cb, b_cb, gx_cb, gy_cb, qx_cb, qy_cb });
+                        pFieldLength = p_cb;
+                    }
+                    else
+                    {
+                        // Match Windows semantics where p,a,b,gx,gy,qx,qy have same length
+                        cbFieldLength = GetMax(new[] { p_cb, a_cb, b_cb, gx_cb, gy_cb, qx_cb, qy_cb });
+                        pFieldLength = cbFieldLength;
+                    }
+
+                    // Match Windows semantics where order and d have same length
+                    int cbSubgroupOrder = GetMax(new[] { order_cb, d_cb });
+
+                    // Copy values to ECParameters
+                    ECParameters parameters = new ECParameters();
+                    parameters.Q = new ECPoint
+                    {
+                        X = Crypto.ExtractBignum(qx_bn, cbFieldLength),
+                        Y = Crypto.ExtractBignum(qy_bn, cbFieldLength)
+                    };
+                    parameters.D = d_cb == 0 ? null : Crypto.ExtractBignum(d_h, cbSubgroupOrder);
+
+                    var curve = parameters.Curve;
+                    curve.CurveType = curveType;
+                    curve.A = Crypto.ExtractBignum(a_bn, cbFieldLength);
+                    curve.B = Crypto.ExtractBignum(b_bn, cbFieldLength);
+                    curve.G = new ECPoint
+                    {
+                        X = Crypto.ExtractBignum(gx_bn, cbFieldLength),
+                        Y = Crypto.ExtractBignum(gy_bn, cbFieldLength)
+                    };
+                    curve.Order = Crypto.ExtractBignum(order_bn, cbSubgroupOrder);
+
+                    if (curveType == ECCurve.ECCurveType.Characteristic2)
+                    {
+                        curve.Polynomial = Crypto.ExtractBignum(p_bn, pFieldLength);
+                    }
+                    else
+                    {
+                        curve.Prime = Crypto.ExtractBignum(p_bn, pFieldLength);
+                    }
+
+                    // Optional parameters
+                    curve.Cofactor = cofactor_cb == 0 ? null : Crypto.ExtractBignum(cofactor_bn, cofactor_cb);
+                    curve.Seed = seed_cb == 0 ? null : Crypto.ExtractBignum(seed_bn, seed_cb);
+
+                    parameters.Curve = curve;
+                    return parameters;
+                }
+            }
+            finally
+            {
+                if (refAdded)
+                    key.DangerousRelease();
+            }
+        }
+
+        private static int GetMax(int[] values)
+        {
+            int max = 0;
```

Comment:
Yes based on context as it's only used for count-of-bytes and is private. If there was a negative for whatever reason return zero is the correct thing here. I'll add the assert however to prevent any future copy-paste errors